dandi/cli/cmd_validate.py

```diff
@@ -84,61 +146,119 @@ def validate(paths, schema=None, devel_debug=False, allow_any_path=False):
     # at this point, so we ignore it although ideally there should be a formal
     # way to get relevant warnings (not errors) from PyNWB
     ignore_benign_pynwb_warnings()
-    view = "one-at-a-time"  # TODO: rename, add groupped
 
-    all_files_errors = {}
-    nfiles = 0
-    for path, errors in validate_(
+    validator_result = validate_(
         *paths,
         schema_version=schema,
         devel_debug=devel_debug,
         allow_any_path=allow_any_path,
-    ):
-        nfiles += 1
-        if view == "one-at-a-time":
-            display_errors(path, errors)
-        all_files_errors[path] = errors
-
-    if view == "groupped":
-        # TODO: Most likely we want to summarize errors across files since they
-        # are likely to be similar
-        # TODO: add our own criteria for validation (i.e. having needed metadata)
-
-        # # can't be done since fails to compare different types of errors
-        # all_errors = sum(errors.values(), [])
-        # all_error_types = []
-        # errors_unique = sorted(set(all_errors))
-        # from collections import Counter
-        # # Let's make it
-        # print(
-        #     "{} unique errors in {} files".format(
-        #     len(errors_unique), len(errors))
-        # )
-        raise NotImplementedError("TODO")
-
-    files_with_errors = [f for f, errors in all_files_errors.items() if errors]
-
-    if files_with_errors:
-        click.secho(
-            "Summary: Validation errors in {} out of {} files".format(
-                len(files_with_errors), nfiles
-            ),
-            bold=True,
-            fg="red",
+    )
+    issues = []
+    for i in validator_result:
+        if not i.severity:
+            continue
+        else:
+            issues.append(i)
+
+    purviews = [
+        list(filter(bool, [i.path, i.path_regex, i.dataset_path]))[0] for i in issues
+    ]
+    if not grouping:
+        display_errors(
+            purviews,
+            [i.id for i in issues],
+            [i.severity for i in issues],
+            [i.message for i in issues],
```

Comment:
> why not to make display_errors just to get a list of them instead of breaking them apart like this just to `zip` them back together inside that function?

Well, the idea was to have a separate function which does as little as possible other than `click.secho()`. The lists need to be populated differently depending on the grouping anyway, and additional edge cases will pop up as we have e.g. multi-path errors, so we do all that in the parent function and just echo depending on the list lengths. If you want I can roll it all up in a single function.

Reply:
I still feel that it is odd design here, not ready ATM to give a complete solution, could at least be `display_errors(purviews, issues)` or alike, with all needed unzipping/zipping done inside if needed.  but ok, we could RF later

Reply:
I'd really suggest leaving it like this for now, and see if moving that code bit to another function is still needed when we merge validation functions.